### PR TITLE
chore: validate incorrect case key for `EnumCoder`

### PR DESCRIPTION
**Pull request by petertonysmith94**
### Summary

- Added validation for EnumCoder to ensure case key is found successfully.
- Added documentation for the error code `INVALID_DECODE_VALUE`.
- Readability changes for enum example docs.

### CODE REVIEW

File: apps/docs/src/guide/types/enums.md [hunk 1]
```diff
@@ -6,40 +6,56 @@ Sway Enums are a little distinct from TypeScript Enums. In this document, we wil
...
 
-<<< @/../../docs-snippets/src/guide/types/enums.test.ts#enum-7{ts:line-numbers}
+<<< @/../../docs-snippets/src/guide/types/enums.test.ts#enum-of-enums-4{ts:line-numbers}
+
+## Errors
+
+While working with enums, you may run into the following issues:
+
+### Using an invalid enum type
+
```
Comment (danielbate) [inline apps/docs/src/guide/types/enums.md:52]:
```suggestion

Thrown when the type being passed to the enum does not match that expected by it.

```

File: apps/docs/src/guide/types/enums.md [hunk 2]
```diff
@@ -6,40 +6,56 @@ Sway Enums are a little distinct from TypeScript Enums. In this document, we wil
...
+<<< @/../../docs-snippets/src/guide/types/enums.test.ts#enum-error-mismatch-type{ts:line-numbers}
+
+### Using an invalid enum value
+
```
Comment (danielbate) [inline apps/docs/src/guide/types/enums.md:56]:
```suggestion

Thrown when the parameter passed is not an expected enum value.

```

File: apps/docs/src/guide/types/enums.md [hunk 3]
```diff
@@ -6,40 +6,56 @@ Sway Enums are a little distinct from TypeScript Enums. In this document, we wil
...
+<<< @/../../docs-snippets/src/guide/types/enums.test.ts#enum-error-value-mismatch{ts:line-numbers}
+
+### Using an invalid enum case key
+
```
Comment (danielbate) [inline apps/docs/src/guide/types/enums.md:60]:
```suggestion

Thrown when the passed enum case is not an expected enum case value.

```

File: apps/docs-snippets/src/guide/types/enums.test.ts [hunk 4]
```diff
@@ -14,36 +15,71 @@ describe(__filename, () => {
...
 
     expect(value).toEqual(enumParam);
-    // #endregion enum-7
+    // #endregion enum-of-enums-4
+  });
+
+  it('should throw when enum value is not the correct type', async () => {
+    // #region enum-error-mismatch-type
+    // Valid values: 'Void', 'Pending', 'Completed'
+    const emumValue = 1;
```
Comment (danielbate) [inline apps/docs-snippets/src/guide/types/enums.test.ts:50]:
```suggestion
    // Valid types: string
    const emumValue: number = 1;
```

**Review by danielbate — COMMENTED** [review]
Looking good, just some thoughts on adding additional context to describe the snippets.

**Review by maschad — COMMENTED** [review]
Looks good, left some suggestions

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by maschad — APPROVED** [review]

**Pull request merged**